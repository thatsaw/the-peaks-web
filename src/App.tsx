import { Default } from "./layouts/Default";
import { Counter } from "./components/Counter";

function App() {
  return (
    <Default>
      <h1>The Peaks</h1>
      <Counter />
    </Default>
  );
}

export default App;
