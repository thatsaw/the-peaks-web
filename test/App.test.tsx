import React from "react";
import { expect, test } from "vitest";
import renderer from "react-test-renderer";
import App from "../src/App";

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();
  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);
  return result as renderer.ReactTestRendererJSON;
}

test("App should render index page", () => {
  const component = renderer.create(<App />);

  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
