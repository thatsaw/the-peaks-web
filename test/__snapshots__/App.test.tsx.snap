// Vitest Snapshot v1

exports[`App should render index page 1`] = `
<div
  className="App"
>
  <div>
    <a
      href="https://vitejs.dev"
      target="_blank"
    >
      <img
        alt="Vite logo"
        className="logo"
        src="/vite.svg"
      />
    </a>
    <a
      href="https://reactjs.org"
      target="_blank"
    >
      <img
        alt="React logo"
        className="logo react"
        src="/src/assets/react.svg"
      />
    </a>
  </div>
  <h1>
    Vite + React
  </h1>
  <div
    className="_card_43ab0b"
  >
    <button
      onClick={[Function]}
    >
      count is 
      0
    </button>
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to test HMR
    </p>
  </div>
  <p
    className="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>
</div>
`;
